nexus/src/context.rs

```diff
@@ -49,6 +49,8 @@ pub struct ServerContext {
     pub external_latencies: LatencyTracker,
     /// registry of metric producers
     pub producer_registry: ProducerRegistry,
+    /// TLS enabled on the external Dropshot server
+    pub external_tls_enabled: bool,
```

Comment:
It feels a little sad to clutter up the context with another config value, but I wasn't sure whether I should get the answer from `nexus` itself because of the `Arc` around it. A method on `nexus` would have to look something like this. @davepacheco?

```rs
pub async fn external_tls_enabled(&self) -> bool {
    let external_server = self.external_server.lock().unwrap().take();
    if let Some(server) = external_server {
        return server.using_tls();
    }
    false
}
```

Reply:
I think it's okay as-is.  One advantage of what's here over that method impl is that if someone asks the question before we've started the external server for some reason, it will report what's configured (which seems right) rather than panicking.

It seems like Dropshot could use a method on the `RequestContext` for this.  Then you could just check that in the handler.